components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/callback/AbstractPasswordCallback.java

```diff
@@ -50,7 +50,12 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
             if (realmConfig == null) {
                 realmConfig = dataHolder.getRealmConfig();
                 if (realmConfig == null) {
-                    realmConfig = getRealmConfig();
+                    // If lazy loading enabled initialize security parameter
+                    if (log.isDebugEnabled()) {
+                        log.debug("Lazy loading security parameters");
+                    }
+                    ServiceComponent.initSecurityParams();
+                    realmConfig = dataHolder.getRealmConfig();
```

Comment:
Shall we define a method for the code segment from line 52-58? I see that the code is duplicated in some other places as well

Reply:
fixed, Thanks